numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2087,6 +2091,11 @@ array_fromfile(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *keywds)
         Py_DECREF(file);
         return NULL;
     }
+    if (npy_fseek(fp, offset, SEEK_CUR) != 0) {
+        PyErr_SetString(PyExc_IOError, "seeking file failed");
```

Comment:
I think this should be `PyErr_SetFromErrno(PyExc_IOError)`

Reply:
fixed